src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/FormPass.php

```diff
@@ -34,12 +34,17 @@ public function process(ContainerBuilder $container)
         $types = array();
 
         foreach ($container->findTaggedServiceIds('form.type') as $serviceId => $tag) {
-            $alias = isset($tag[0]['alias'])
-                ? $tag[0]['alias']
-                : $serviceId;
+            // The following if-else block is deprecated and can be removed
+            // in Symfony 3.0
+            if (isset($tag[0]['alias'])) {
+                $types[$tag[0]['alias']] = $serviceId;
+            } else {
+                $types[$serviceId] = $serviceId;
```

Comment:
@Tobion it would not work for the case of @DavidBadura anyway, because Symfony validates that the ``getName`` method returns the same value than the one used when registering the tag lazily (to be sure we don't create weird things when using the name later while we started with the registered alias)